service/lib/dinstaller/manager.rb

```diff
@@ -43,106 +44,115 @@ module DInstaller
-
-      # call inst bootloader to get properly initialized bootloader
-      # sysconfig before package installation
-      Yast::WFM.CallFunction("inst_bootloader", [])
-
-      progress.next_step("Installing Software")
-      software.install
+    def install_phase
+      installation_phase.install
+
+      start_progress(9)
+
+      progress.step("Reading software repositories") do
+        software.probe
+        Yast::Installation.destdir = "/mnt"
+      end
+
+      progress.step("Partitioning") do
+        # lets propose it here to be sure that software proposal reflects product selection
+        # FIXME: maybe repropose after product selection change?
+        # first make bootloader proposal to be sure that required packages are installed
+        proposal = ::Bootloader::ProposalClient.new.make_proposal({})
+        logger.info "Bootloader proposal #{proposal.inspect}"
+        storage.install
+        # propose software after /mnt is already separated, so it uses proper
+        # target
+        software.propose
+
+        # call inst bootloader to get properly initialized bootloader
+        # sysconfig before package installation
+        Yast::WFM.CallFunction("inst_bootloader", [])
+      end
+
+      progress.step("Installing Software") { software.install }
 
       on_target do
-        progress.next_step("Writing Users")
-        users.write(progress)
+        progress.step("Writing Users") { users.write }
 
-        progress.next_step("Writing Network Configuration")
-        network.install(progress)
+        progress.step("Writing Network Configuration") { network.install }
 
-        progress.next_step("Installing Bootloader")
-        security.write(progress)
-        ::Bootloader::FinishClient.new.write
+        progress.step("Installing Bootloader") do
+          security.write
+          ::Bootloader::FinishClient.new.write
+        end
 
-        progress.next_step("Saving Language Settings")
-        language.install(progress)
+        progress.step("Saving Language Settings") { language.install }
 
-        progress.next_step("Writing repositories information")
-        software.finish
+        progress.step("Writing repositories information") { software.finish }
 
-        progress.next_step("Finishing installation")
-        finish_installation
+        progress.step("Finishing installation") { finish_installation }
       end
-
-      progress.next_step("Installation Finished")
-      status_manager.change(Status::Installed.new)
     end
     # rubocop:enable Metrics/AbcSize
 
-    # Software manager
+    # Software client
     #
-    # @return [Software]
+    # @return [DBus::Clients::Software]
     def software
-      @software ||= DBus::Clients::Software.new
+      @software ||= DBus::Clients::Software.new.tap do |client|
+        client.on_product_selected do |selected|
+          config.pick_product(selected)
+          config_phase
+        end
+        client.on_service_status_change do |status|
```

Comment:
I have discovered that having two calls to `on_signal` will not work. The [second one](https://github.com/joseivanlopez/d-installer/blob/06ec911c9003f16a51c78c0bb950a1b5f7608ca1/service/lib/dinstaller/dbus/clients/with_service_status.rb#L50) will overwrite [the first one](https://github.com/joseivanlopez/d-installer/blob/06ec911c9003f16a51c78c0bb950a1b5f7608ca1/service/lib/dinstaller/dbus/clients/software.rb#L164). So we might need to implement this in a slightly different way (registering the callbacks and listening just once).